### DISCUSSION TITLE
[Metal] Want to know how createSwapChainS is triggered and called on the Metal platform？

### DISCUSSION THREAD

**Discussion by ccworld1000** [google/filament#6162] [Q&A]
Want to know how createSwapChainS is triggered and called on the Metal platform？

I searched for documents, codes, and problems, as if this detail had not been specifically mentioned.
Or whether I overlooked the details

**Comment by ccworld1000** [comment]
1、mSwapChain = engine.getDriverApi().createSwapChain(nativeWindow, flags);
2、MetalDriver::createSwapChainS

Or, in another way, ask how the above two are related.

**Reply by bejado** [reply]
`createSwapChainS` is called synchronously when you call `engine.getDriverApi().createSwapChain(...)`. Its job is just to allocate memory and return a handle. At the same time, a call to `createSwapChainR` is added to the driver command queue, which will be invoked asynchronously on the driver thread with the actual arguments.
